bin/tx-decoder/src/app.rs

```diff
@@ -128,8 +126,8 @@ impl<'a> AppState<'a> {
 	fn decode(decoder: &Decoder, block: BlockModel, spec: SpecVersion, errors: &mut Vec<String>) -> Result<(), Error> {
 		log::debug!("Decoding block {}, spec_version {}, ext length {}", block.block_num, spec, block.ext.len());
 		match decoder.decode_extrinsics(spec, &block.ext) {
-			Err(e) => {
-				let e = e.context(format!("Failed to decode block {}", block.block_num));
+			e @ Err(_) => {
+				let e = e.context(format!("Failed to decode block {}", block.block_num)).unwrap_err();
```

Comment:
I'm curious to know why this change was necessary? does the `context` method on the trait give you something that the `context` method on `anyhow::Error` doesn't?

Reply:
since i switched tx-decoder to use the facade crate, the method returns a thiserror::Error which doesn't have a direct `context` method like `anyhow::Error` does. Instead the context method is accessed through the `Context` trait on the general `Result` type

Reply:
Aah ok, so you're using the trait to turn it into an `anyhow::Error` with context!

I did have a look and found that you could perhaps do something like:

`anyhow::Error::new(e).context(format!("Failed to decode block {}", block.block_num))` to avoid the result unwrapping thing if you like.